docs/NNNN-n-tuple-omit-prefix-storage-layout.md

```diff
@@ -0,0 +1,183 @@
+# OCFL Community Extension NNNN: N Tuple Omit Prefix Storage Layout
+
+  * **Extension Name:** NNNN-n-tuple-omit-prefix-storage-layout
+  * **Authors:** Michael Vandermillen and Andrew Woods
+  * **Minimum OCFL Version:** 1.0
+  * **OCFL Community Extensions Version:** 1.0
+  * **Obsoletes:** n/a
+  * **Obsoleted by:** n/a
+
+## Overview
+
+This storage root extension describes an OCFL storage layout combining a pairtree-like root directory structure derived from prefix-omitted object identifiers, followed by the prefix-omitted object identifier themselves. 
+The OCFL object identifiers are expected to contain prefixes which are removed in the mapping to directory names. 
+The OCFL object identifier prefix is defined as all characters before and including a configurable delimiter.
+Where the prefix-omitted identifier length is less than tuple size * number of tuples, the remaining object id (prefix omitted) is left or right-side, zero-padded (configurable, left default), or not padded (none), and optionally reversed (default false).
+The object id is then divided into N n-tuple segments, and used to create nested paths under the OCFL storage root, followed by the prefix-omitted object id directory.  
+
+This layout combines the advantages of 0006-flat-omit-prefix-storage-layout (directory name transparency) and the 0004-hashed-n-tuple-storage-layout (enhanced file system/bucket performance). 
+
+The limitations of this layout are filesystem dependent, but are generally as follows:
+
+* The size of object identifiers, minus the length of the prefix, cannot exceed the maximum allowed directory name size
+  (eg. 255 characters)
+* Object identifiers cannot include characters that are illegal in directory names
+
+## Parameters
+
+### Summary
+
+* **Name:** `delimiter`
+  * **Description:** The case-insensitive, delimiter marking the end of the OCFL object identifier prefix; MUST consist
+    of a character string of length one or greater. If the delimiter is found multiple times in the OCFL object
+    identifier, its last occurence (right-most) will be used to select the termination of the prefix.
+  * **Type:** string
+  * **Constraints:** Must not be empty
+  * **Default:** :
+* **Name**: `tupleSize`
+  * **Description:** Indicates the segment size (in characters) to split the
+    identifier into
+  * **Type:** number
+  * **Constraints:** An integer between 1 and 32 inclusive
+  * **Default:** 3
+* **Name:** `numberOfTuples`
+  * **Description:** Indicates the number of segments to use for path generation
+  * **Type:** number
+  * **Constraints:** An integer between 1 and 32 inclusive
+  * **Default:** 3
+* **Name:** `zeroPadding`
+  * **Description:** Indicates whether to use left or right zero padding for ids less than tupleSize * numberOfTuples
+  * **Type:** string
+  * **Constraints:** Must be either "left", "right" or "none"
```

Comment:
I don't think the case of "none" combined with a prefix stripped id of fewer then N*n characters is properly explained in the procedure step 4. Is "none" really a useful option?

Reply:
In the case that "tupleSize=3" and "numberOfTuples=2", and the prefix-stripped ID is: 12345, it seems like it is helpful to leave the option to not use zero-padding. Is there a more clear way of indicating such a desire in the configuration than using "none"?

Reply:
@awoods -- what is the expected result for the example you give? Is it `123/45/12345` (with reverse=false)?

Reply:
@zimeon : yes. With no zero-padding, the expectation in this case would be as you say: `123/45/12345`

Reply:
`none` would also have a problem with something like `123` and `123123`, which would result in the clashing object root paths `123/123` and `123/123/123123`.

Reply:
Good point @pwinckles !

Reply:
Thanks all, I'll get rid of the none option and push

Reply:
I have removed the 2 mentions of "none" from the spec